### PR TITLE
Change @Concurrent annotation name to @strand

### PR DESCRIPTION
**Pull request by vinok88**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal [hunk 1]
```diff
@@ -34,12 +34,27 @@ type ArgsData record {|
 # Defaultable annotation data generated at compile time. This is for internal use.
 annotation ArgsData DefaultableArgs on function;
 
-# Defines a disptcher to be used for concurrent execution of strands.
+//# Defines a disptcher to be used for concurrent execution of strands.
```
Comment (hasithaa) [inline langlib/lang.annotations/src/main/ballerina/src/lang.annotations/annotations.bal]: Shall we remove commented code?

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/AnnotationUtils.java [hunk 2]
```diff
@@ -96,11 +96,16 @@ public static void processFPValueAnnotations(FPValue fpValue, MapValue globalAnn
...
      * @param fpValue function pointer to be invoked
      * @return true if should run concurrently
      */
     public static boolean isConcurrent(FPValue fpValue) {
-        return  ((BFunctionType) fpValue.getType()).getAnnotation("ballerina/lang.annotations",
-                "concurrent") != null ? true : false;
+        Object value = ((BFunctionType) fpValue.getType()).getAnnotation("ballerina/lang" +
+                ".annotations", "strand");
+        if (value != null) {
+            return ((MapValue) value).get("thread") == "any";
```
Comment (SupunS) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/AnnotationUtils.java]: Shall we move these to constants?

File: compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_terminator_gen.bal [hunk 3]
```diff
@@ -681,24 +681,26 @@ type TerminatorGenerator object {
...
-                v.moduleId.org == "ballerina" && 
-                v.moduleId.name == "lang.annotations") {
-                    concurrent = true;
-                    if (v.annotValues.length() > 0) {
-                        if (v.annotValues[0] is ()){
-                            break;
+                    v.annotTagRef.value == "strand" && 
+                    v.moduleId.org == "ballerina" && 
+                    v.moduleId.name == "lang.annotations") {
+                        if (v.annotValues.length() > 0) {
```
Comment (SupunS) [inline compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_terminator_gen.bal]: shall we invert the condition?

**Comment by vinok88** [comment]
Please go ahead and merge this, review suggestions will be handled in a separate PR.

**Pull request merged**